utils/src/main/kotlin/Utils.kt

```diff
@@ -59,18 +58,21 @@ const val PARAMETER_ORDER_LOGGING = 2
 const val PARAMETER_ORDER_HELP = 100
 
 /**
- * Check whether the specified two or more collections have no elemets in common.
+ * Return the set of elements the specified two or more collections have in common, or an empty set if all collections
+ * are disjoint.
  */
-fun disjoint(c1: Collection<*>, c2: Collection<*>, vararg cN: Collection<*>): Boolean {
+fun <T> disjoint(c1: Collection<T>, c2: Collection<T>, vararg cN: Collection<T>): Set<T> {
     val c = listOf(c1, c2, *cN)
 
+    val commonElememts = mutableSetOf<T>()
```

Comment:
I'm not sure about the sizes of the collection usually passed to this function,
if sizes are not small consider making it more efficient, like:
```
val list = listOf(c2, *cN)
var commonElements = c[a]

for (i in list.indices) {
    commonElements = commonElements.intersect(list[i])
}
```

Reply:
See above, this computes something different.